service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/EncounterExtractor.java

```diff
@@ -0,0 +1,52 @@
+package uk.nhs.adaptors.gp2gp.ehr.utils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.ListResource;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+
+public final class EncounterExtractor {
+    private static final String CONSULTATION_LIST_CODE = "1149501000000101";
+
+    private EncounterExtractor() {
+    }
+
+    public static List<String> extractEncounterReferences(List<Bundle.BundleEntryComponent> entries) {
```

Comment:
maybe extractEncounterReferencesFromConsultationList?

Reply:
extractEncounterReferencesFromEncounterList would be a better name

Even though the display is "List of consultations" there is yet another type of list we'll see in NIAD-904 named "Consultation" which is referred to as the List(Consultation) https://developer.nhs.uk/apis/gpconnect-1-5-0/accessrecord_structured_development_list_consultation.html